src/embind/embind.js

```diff
@@ -2147,6 +2147,363 @@ var LibraryEmbind = {
     });
   },
 
+  $init_RegisteredTrivialObject__deps: [
+    '$RegisteredTrivialObject_fromWireType', '$RegisteredTrivialObject_toWireType'],
+  $init_RegisteredTrivialObject: function() {
+    RegisteredTrivialObject.prototype['argPackAdvance'] = 8;
+    RegisteredTrivialObject.prototype['readValueFromPointer'] = function(){ throw ":TODO: readValueFromPointer";};
+    RegisteredTrivialObject.prototype['fromWireType'] = RegisteredTrivialObject_fromWireType;
+    RegisteredTrivialObject.prototype['toWireType'] = RegisteredTrivialObject_toWireType;
+  },
+
+  $RegisteredTrivialObject__deps: [
+    '$init_RegisteredTrivialObject'],
+  $RegisteredTrivialObject__postset: 'init_RegisteredTrivialObject()',
+  $RegisteredTrivialObject: function(
+    name,
+    size,
+    registeredClass,
+    isReference,
+    isConst
+  ) {
+    this.name = name;
+    this.size= size;
+    this.registeredClass = registeredClass;
+    this.isReference = isReference;
+    this.isConst = isConst;
+  },
+
+  $RegisteredTrivialObject_fromWireType__deps: [
+     '$makeTrivialClassHandle'],
+  $RegisteredTrivialObject_fromWireType: function(ptr) {
+
+    var buffer = HEAPU32['buffer'];
+    var objectBuffer = buffer.slice( ptr, ptr + this.size );
+
+    var o = makeTrivialClassHandle(this.registeredClass.instancePrototype, objectBuffer);
+
+    //C++ wire is making copy, so free it here
+    this.registeredClass.rawDestructor(o);
+    return o;
+  },
+
+  $RegisteredTrivialObject_copyAndFreePtr: function(a){
+      var ptr = a[1];
+      var size = a[2];
+      a[0].$$.buffer = HEAP32['buffer'].slice(ptr, ptr + size);
+      _free(ptr);
+  },
+  $RegisteredTrivialObject_freePtr: function(ptr){
+      _free(ptr);
+  },
+  $RegisteredTrivialObject_toWireType__deps:[
+      '$RegisteredTrivialObject_copyAndFreePtr', '$RegisteredTrivialObject_freePtr'],
+  $RegisteredTrivialObject_toWireType: function(destructors, object) {
+
+      //:TODO: verify alignment
+    var local_heap = new Uint32Array( object.$$.buffer );
+    var ptr = _malloc( this.size );
+    var heap = HEAPU32;
+
+    for(i=0; i< this.size; i+=4){
```

Comment:
If you don't mind, could you update the code to match the rest of the file in terms of whitespace?